Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -13,9 +13,32 @@ import Foundation
 @objc
 open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate {
     var interceptor: URLSessionInterceptorType?
+    let firstPartyURLsFilter: FirstPartyURLsFilter?
 
     @objc
     override public init() {
+        firstPartyURLsFilter = nil
+        interceptor = URLSessionAutoInstrumentation.instance?.interceptor
+        if interceptor == nil {
+            let error = ProgrammerError(
+                description: """
+                `Datadog.initialize()` must be called before initializing the `DDURLSessionDelegate` and
+                first party hosts must be specified in `Datadog.Configuration`: `track(firstPartyHosts:)`
+                to enable network requests tracking.
+                """
+            )
+            consolePrint("\(error)")
+        }
```

Comment:
Instead of duplicating this error code, we can move the validation to private static function:
```swift
private static func retrieveInterceptor() -> URLSessionInterceptorType? {
   guard let interceptor = URLSessionAutoInstrumentation.instance?.interceptor else {
      let error = ProgrammerError(
         description: """
            `Datadog.initialize()` must be called before initializing the `DDURLSessionDelegate` and
            first party hosts must be specified in `Datadog.Configuration`: `track(firstPartyHosts:)`
            to enable network requests tracking.
            """
         )
         consolePrint("\(error)")
         return nil
   }
   return interceptor
}
```